File: memory_test/logn/logn.cpp
#include <iostream>
#include <vector>
#include <chrono>
#include <iomanip>
#include </home/teaguy21/Documents/memory_test/constant/constant.h>
using namespace std;

// An iterative binary search function.
int binarySearch(vector<int> arr, int low, int high, int x)
{
    while (low <= high) {
        int mid = low + (high - low) / 2;

        if (arr[mid] == x)
            return mid;

        if (arr[mid] < x)
            low = mid + 1;

        else
            high = mid - 1;
    }
    return -1;
}

// Driver code
int main(void)
{
    vector<int> arr;
    int amount = N_VALUE; 
    for (int i = 1; i <= amount; i++)
    {
        arr.push_back(i);
    }
    int x = 1;
    int n = arr.size();

    auto start = chrono::high_resolution_clock::now();
    int result = binarySearch(arr, 0, n - 1, x);
    auto end = chrono::high_resolution_clock::now();

    auto duration = chrono::duration_cast<chrono::nanoseconds>(end - start);
    cout << fixed << setprecision(9) << duration.count() / 1000000000.0 << endl;
    // (result == -1)
    //     ? cout << "Element is not present in array"
    //     : cout << "Element is present at index " << result;
    return 0;
}

File: plotting/Python/logn.py
import time


def binarySearch(arr, low, high, x):

    while low <= high:

        mid = low + (high - low) // 2

        if arr[mid] == x:
            return mid

        elif arr[mid] < x:
            low = mid + 1

        else:
            high = mid - 1

    return -1

with open("/home/teaguy21/Documents/plotting/constant.txt", "r") as file:
  first_line = file.readline()
  number_string = first_line.rstrip("\n")
  number = int(number_string)

arr = []
for i in range(number):
    arr.append(i)
    i += 1

x = 1

start = time.time_ns()
result = binarySearch(arr, 0, len(arr) - 1, x)
end = time.time_ns()   
print(format((end - start) / 10 ** 9, '.6f'))
# if result != -1:
#     print("Element is present at index", result)
# else:
#     print("Element is not present in array")
